File: clients/python/src/cien_agent_sdk/admin/environments.py
from __future__ import annotations

from typing import Any

from ..base import EndpointGroup
from ..utils import drop_none


class AdminEnvironmentsAPI(EndpointGroup):
    """/api/admin/environments endpoints."""

    def list(self, *, coid: str, include_sync: bool = False) -> dict[str, Any]:
        return self._get("/api/admin/environments", params={"coid": coid, "include_sync": include_sync})

    def get(self, coid: str, *, environment: str = "staging", include_sync: bool = False) -> dict[str, Any]:
        return self._get(
            f"/api/admin/environments/{coid}",
            params={"environment": environment, "include_sync": include_sync},
        )

    def create(self, *, data: dict[str, Any], environment: str = "staging") -> dict[str, Any]:
        return self._post(
            "/api/admin/environments",
            json={"data": data},
            params={"environment": environment},
        )

    def update(
        self,
        coid: str,
        *,
        updates: dict[str, Any],
        environment: str = "staging",
    ) -> dict[str, Any]:
        return self._patch(
            f"/api/admin/environments/{coid}",
            json={"updates": updates},
            params={"environment": environment},
        )

    def delete(self, coid: str, *, environment: str = "staging") -> dict[str, Any]:
        return self._delete(f"/api/admin/environments/{coid}", params={"environment": environment})

    def copy(
        self,
        coid: str,
        *,
        source_environment: str = "prod",
        destination_environment: str = "staging",
        include_sync: bool = True,
        overwrite_sync: bool = True,
    ) -> dict[str, Any]:
        return self._post(
            f"/api/admin/environments/{coid}/copy",
            json=drop_none(
                {
                    "source_environment": source_environment,
                    "destination_environment": destination_environment,
                    "include_sync": include_sync,
                    "overwrite_sync": overwrite_sync,
                }
            ),
        )

File: clients/node/src/admin/environments.ts
import { EndpointGroup } from '../base.js'
import { dropNullish } from '../utils.js'

export class AdminEnvironmentsAPI extends EndpointGroup {
  list(params: { coid: string; include_sync?: boolean, include_config?: boolean }) {
    return super.requestGet<Record<string, unknown>>('/api/admin/environments', {
      params: { coid: params.coid, include_sync: params.include_sync ?? false, include_config: params.include_config ?? false },
    })
  }

  get(coid: string, params: { environment?: string; include_sync?: boolean, include_config?: boolean } = {}) {
    return super.requestGet<Record<string, unknown>>(`/api/admin/environments/${coid}`, {
      params: {
        environment: params.environment ?? 'staging',
        include_sync: params.include_sync ?? false,
        include_config: params.include_config ?? false,
      },
    })
  }

  create(payload: { data: Record<string, unknown>; environment?: string }) {
    return this.requestPost<Record<string, unknown>>('/api/admin/environments', {
      params: { environment: payload.environment ?? 'staging' },
      json: { data: payload.data },
    })
  }

  update(coid: string, payload: { updates: Record<string, unknown>; environment?: string }) {
    return this.requestPatch<Record<string, unknown>>(`/api/admin/environments/${coid}`, {
      params: { environment: payload.environment ?? 'staging' },
      json: { updates: payload.updates },
    })
  }

  delete(coid: string, params: { environment?: string } = {}) {
    return super.requestDelete<Record<string, unknown>>(`/api/admin/environments/${coid}`, {
      params: { environment: params.environment ?? 'staging' },
    })
  }

  copy(
    coid: string,
    payload: {
      source_environment?: string
      destination_environment?: string
      include_sync?: boolean
      overwrite_sync?: boolean
    } = {}
  ) {
    return this.requestPost<Record<string, unknown>>(`/api/admin/environments/${coid}/copy`, {
      json: dropNullish({
        source_environment: payload.source_environment ?? 'prod',
        destination_environment: payload.destination_environment ?? 'staging',
        include_sync: payload.include_sync ?? true,
        overwrite_sync: payload.overwrite_sync ?? true,
      }),
    })
  }
}
